kernel/src/table_features/mod.rs

```diff
@@ -124,6 +128,422 @@ pub(crate) enum FeatureType {
+static CHANGE_DATA_FEED_INFO: FeatureInfo = FeatureInfo {
+    name: "changeDataFeed",
+    min_reader_version: 1,
+    min_writer_version: 4,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::NotSupported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_change_data_feed == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static GENERATED_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "generatedColumns",
+    min_reader_version: 1,
+    min_writer_version: 4,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IDENTITY_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "identityColumns",
+    min_reader_version: 1,
+    min_writer_version: 6,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IN_COMMIT_TIMESTAMP_INFO: FeatureInfo = FeatureInfo {
+    name: "inCommitTimestamp",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_in_commit_timestamps == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static ROW_TRACKING_INFO: FeatureInfo = FeatureInfo {
+    name: "rowTracking",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    required_features: &[FeatureRequirement::Supported(TableFeature::DomainMetadata)],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| props.enable_row_tracking == Some(true)),
+};
+
+#[allow(dead_code)]
+static DOMAIN_METADATA_INFO: FeatureInfo = FeatureInfo {
+    name: "domainMetadata",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
+    conflicting_features: &[],
+    read_support: SupportCheck::NotSupported,
+    write_support: SupportCheck::Supported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static ICEBERG_COMPAT_V1_INFO: FeatureInfo = FeatureInfo {
+    name: "icebergCompatV1",
+    min_reader_version: 1,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    required_features: &[],
```

Comment:
this one has requirements too:

> * Require that Column Mapping be enabled and set to either name or id mode
> * Require that Deletion Vectors are not supported (and, consequently, not active, either). i.e., the deletionVectors table feature is not present in the table protocol.
> * Block adding Map/Array/Void types to the table schema (and, thus, block writing them, too)

Especially the iceberg ones may require that the check also take metadata/schema. (Ex: block map/array/void types). I'm unsure if this is out of scope, but I'd like if we could just handle all the checks in one place.  What do you think?